src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const keys = items.map((item) => item[key]);
+  const uniqueKeys = Array.from(new Set(keys));
+  const res: GroupsMap<T> = {};
+
+  uniqueKeys.forEach((uniqueKey) => {
+    res[uniqueKey as string] = items.filter((item) => {
```

Comment:
Casting `uniqueKey` to a string using `as string` might not be safe if the `key` is not of type `string`. Consider ensuring that the keys are strings or handle other types appropriately.